colossus-tests/src/test/scala/colossus/controller/InputControllerSpec.scala

```diff
@@ -60,6 +62,17 @@ class InputControllerSpec extends ColossusSpec with CallbackMatchers{
       //now it should be done
       executed must equal(true)
     }
+    
+    "reject data above the size limit" in {
+      val input = ByteString("hello")
+      val config = ControllerConfig(4, 50.milliseconds, "test-controller", 2L)
+      val con = static(config)
+      intercept[ParseException] {
```

Comment:
Do we want to take this time and change this exception to something more specific. Technically, this is not a parsing issue, its a data size issue.  I would vote we throw a different type of exception specific to the actual cause of too large input.